### PR TITLE
No more screenshots

### PR DESCRIPTION
**Pull request by LWinterberg**
Resolves: #5718

Also removes some scripts relying on the screenshot tool, and VetoDialogHook which - as far as I can tell - only was useful for screenshots.

### CODE REVIEW

**Comment by petersampsonaudacity** [comment]
Testing on W10 with @LWinterberg 's branch build for this:  

This looks to be fixed in Leo's branch.

There is no **Screenshot** entry in the Tools menu
image

I have added P1s to the Manual for this for when this gets pulled into master for 3.5.0

File: scripts/piped-work/get_gui_structure.py [hunk 1]
```diff
@@ -65,7 +65,6 @@ def getStructure() :
     #do( 'Help: CommandName=Help' )
     #do( 'Help: CommandName=SetPreference' )
     #do( 'SetPreference: PrefName=GUI/Theme PrefValue=light' )
-    #do( 'Screenshot: CaptureMode=menus' )
```
Comment (Paul-Licameli) [inline scripts/piped-work/get_gui_structure.py:68]: Wow, I never once in my life looked in this subdirectory of the source tree before!

Did you verify that things still run?

What remains in here can serve as examples for how to use the script pipe.

Comment (Paul-Licameli) [reply scripts/piped-work/get_gui_structure.py:68]: I'm trying to run the old Perl script.  What I didn't know is that first you must start Audacity and have an audio track, not an empty project.  I get frequent error dialogs, and now the new export dialog messes up the macros by pausing for user input.

So anyway making this all run again seems out of scope for now.  But tell @dozzzzer and @chinakov to study these examples if they haven't.  Maybe writing Python programs driving the script pipe for regression tests would be easier than writing macros using Audacity's own user interface.

Comment (Paul-Licameli) [reply scripts/piped-work/get_gui_structure.py:68]: I see `pipe_test.py` too, and there is still mention of screenshot in a comment in it.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -891,7 +891,6 @@ list( APPEND SOURCES
       widgets/TimeFormat.h
       widgets/UnwritableLocationErrorDialog.cpp
       widgets/UnwritableLocationErrorDialog.h
-      widgets/VetoDialogHook.h
```
Comment (Paul-Licameli) [inline src/CMakeLists.txt:898]: You are correct that `VetoDialogHook.h` existed only to enable screenshots.  This commit looks complete to me, for the purpose of removing it.

**Comment by petersampsonaudacity** [comment]
> and it was used to generate pictures in the online Audacity manual.

_**NOT**_ by me it wasn't - and after Gale Andrews died I was the major contributor and editor of the manual.

It was never that reliable a tool - I used my PC screenshot button and edited in Paint.

>But I would ask first why it's important to remove, rather than letting this harmless thing remain in Audacity 3.

Because basically it's just useless clutter that burdens the Tools menu

**Comment by Paul-Licameli** [comment]
> > and it was used to generate pictures in the online Audacity manual.
> 
> _**NOT**_ by me it wasn't - and after Gale Andrews died I was the major contributor and editor of the manual _(I don't know about Steve, Bill or Ed)_
> 
> I found that it was never that reliable a tool - I used my PC screenshot button and edited in Paint.
> 
> > But I would ask first why it's important to remove, rather than letting this harmless thing remain in Audacity 3.
> 
> Because basically it's just useless clutter that burdens the Tools menu
> 
> **Big** +1 from me for its removal

In which case, I'll formally approve the PR.

**Review by Paul-Licameli — APPROVED** [review]
Reminder to fix just one comment in Python.

**Comment by petersampsonaudacity** [comment]
It's also worth removing becase its UI is so rubbishy - it lets Audacity down.

See:
image

Many years ago there was an ENH to improve/upgrade it - but no-one ever picked that up, so it just languished ...

**Comment by Paul-Licameli** [comment]
> > But I would ask first why it's important to remove, rather than letting this harmless thing remain in Audacity 3.
> 
> It has a number of bugs associated with it (see #2335), is not maintained, won't make the jump to 4, and crucially, is not used. Peter doesn't use it, QA doesn't want it, and users have easier to use options these days (Win+Shift+S on Windows, Cmd+Shift+4 on Mac). If there was some internal use for it, I'd leave the short-format dialog and the functionality intact, and just remove the button-rich dialog, but there doesn't appear to be one.

I already approved!

I’m convinced!

Thank you for yet more reasons!

While I default to conservative behaviorally neutral refactorings, and don’t make redesign my job, this thing has also been a mildly annoying maintenance burden sometimes.  Bye bye.

**Pull request merged**